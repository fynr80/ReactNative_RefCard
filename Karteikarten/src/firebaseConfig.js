
import firebase from 'firebase/compat/app';

const firebaseConfig = {
    apiKey: "AIzaSyBksswFLr1YB6en-sRf8mqSFle7bhAzSgU",
    authDomain: "swtp-812df.firebaseapp.com",
    databaseURL: "https://swtp-812df-default-rtdb.firebaseio.com",
    projectId: "swtp-812df",
    storageBucket: "swtp-812df.appspot.com",
    messagingSenderId: "44236558384",
    appId: "1:44236558384:web:2f15f6b5fb190634b71df5",
    measurementId: "G-W33L0G7D18"
};
firebase.initializeApp(firebaseConfig);
export {firebase};
