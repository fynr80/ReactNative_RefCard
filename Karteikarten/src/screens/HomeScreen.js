/**
 * @file HomeScreen of RefCard app
 * @author Fatih Yanar <fatih.yanar@mni.thm.de>
 */


import * as React from 'react';
import {
    View,
    Text,
    StyleSheet,
    TouchableOpacity,
    FlatList,
    Image}
from 'react-native';
import {useEffect,useState} from "react";

import {SafeAreaView} from 'react-native-safe-area-context';

//import {LinearGradient} from "expo-linear-gradient";

import SetCategories from "../components/SetCategories";

import firebase from 'firebase/compat/app';

export default function HomeScreen({ }) {

    const [buttonFlag, setButtonFlag] = useState(true);
    const [myCategoriesArr, setMyCategoriesArr] = useState([]);


    useEffect(() => {
        getCategories();
    }, [])

    const setButtonFlag_func = () => {
        if (!buttonFlag) setButtonFlag(true)
    }

    const setButtonFlag_func2 = () => {
        if (buttonFlag) setButtonFlag(false)
    }

    /**
     * Get all Categories and save them into the Array to be displayed
     */
    const getCategories = () => {
        firebase.firestore()
            .collection('category')
            .onSnapshot(snapshot => {
            const arr = snapshot.docs.map((doc) => ({
                id: doc.id,
                name: doc.data().titel,
                categoryColor:doc.data().color
            }))
            setMyCategoriesArr(arr)
        })
    }


    /**
     * Toggles one of the two Buttons when Pressed
     * @param buttonFlag : Boolean
     * @returns Style of the Button
     */
    const btnStyle = function (buttonFlag) {
        if (buttonFlag) {
            return [styles.twoButtonFlagTrue]
        }
        return [styles.twoButtonFlagFalse]
    }


    return (
        <SafeAreaView style={styles.container}>
            <View style={styles.header}>
                <Image
                    style={{height:90,width:120,backgroundColor:'white'}}
                    source={require('../asserts/Logonew.png')}
                />
            </View>
            <View style={styles.lastOpenedTextContainer}>
                <Text style={styles.lastOpenedText} testID={"LastOpenedText"}>Last Opened</Text>
            </View>
            <TouchableOpacity  style={styles.lastOpenedItems}>
                <View style={styles.lastItem}>
                {/*<LinearGradient style={styles.lastItem} colors={['#2B32B2','#1488CC']}>*/}
                    <Text style={[styles.lastOpenedText,{color: 'white',paddingLeft:10,}]}>Diskrete Mathematik</Text>
                    <Text style={[styles.lastOpenedText,{color: 'white',paddingLeft:10,fontSize: 15,fontWeight: 'normal',opacity:0.8}]}>15 Questions</Text>
                {/*</LinearGradient>*/}
                </View>
            </TouchableOpacity>
            <View style={styles.buttons}>
                <TouchableOpacity testID={"MyCatBtn"} onPress={setButtonFlag_func}
                                  style={btnStyle(buttonFlag)}>
                    <Text testID={"MyCatText"} style={[styles.lastOpenedText,{color: buttonFlag ? '#F6F6F6': 'black',textAlign:'center'}]}>My Category</Text>
                </TouchableOpacity>
                <TouchableOpacity  testID={"AllCatBtn"} onPress={setButtonFlag_func2}
                                   style={btnStyle(!buttonFlag)}>
                    <Text testID={"AllCatText"} style={[styles.lastOpenedText,{color: buttonFlag ? 'black': '#F6F6F6',  textAlign:'center'}]}>All Category</Text>
                </TouchableOpacity>
            </View>
            <View style={styles.catList}>
                <View style={styles.categoryItem}>
                    <FlatList
                        testID={"ItemsList"}
                        numColumns={1}
                        data={myCategoriesArr}
                        keyExtractor={(item, index) => item + index}
                        renderItem={({item}) => <SetCategories name={item.name} id={item.id} categoryColor={item.categoryColor}/>}
                    />
                </View>
            </View>
        </SafeAreaView>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignContent:'flex-start',
        backgroundColor:'white',
    },
    header:{
        backgroundColor:'white',
        flexDirection:'row',
        flex:1.9,
        alignItems:'center',
        paddingLeft:2,
        paddingRight:2,
        justifyContent:'center',
        marginTop:3
    },
    headerText:{
        fontSize:20,
        fontWeight:'bold',
        marginLeft:90
    },
    lastOpenedTextContainer: {
        backgroundColor:'white',
        flex:1,
        alignItems:'center',
        justifyContent:'flex-start',
        flexDirection:'row',
        paddingLeft:10
    },
    lastOpenedText: {
        fontSize:20,
        fontWeight:'bold',
        color:'black'
    },
    lastOpenedItems: {
        backgroundColor:'white',
        flexDirection:'row',
        flex:2,
        alignItems:'center',
        justifyContent:'space-between',
        paddingLeft:10,
        paddingRight:10,
        width:'100%'
    },
    lastItem: {
        justifyContent:'center',
        width:'100%',
        height:'80%',
        backgroundColor:'blue',
        borderRadius:10,
    },
    buttons:{
        backgroundColor:'white',
        flexDirection:'row',
        flex:2,
        alignItems:'center',
        justifyContent:'space-between',
        paddingLeft:10,
        paddingRight:10
    },
    twoButtonFlagTrue: {
        justifyContent:'center',
        width:'46%',
        height:'60%',
        backgroundColor:'#FD4365',
        borderRadius:30,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 7,
        },
        shadowOpacity: 0.43,
        shadowRadius: 9.51,
        elevation: 15,
    },
    twoButtonFlagFalse: {
        justifyContent:'center',
        width:'46%',
        height:'60%',
        backgroundColor:'#FBF9F9',
        borderRadius:30,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 7,
        },
        shadowOpacity: 0.43,
        shadowRadius: 9.51,
        elevation: 15,
    },
    catList:{
        backgroundColor:'white',
        flexDirection:'column',
        flex:8,
        alignItems:'center',
        justifyContent:'space-between',
        paddingLeft:10,
        paddingRight:10,
    },
    categoryItem :{
        backgroundColor:'white',
    }
});
