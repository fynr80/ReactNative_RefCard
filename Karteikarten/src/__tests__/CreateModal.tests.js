import React from 'react';
import {fireEvent, render} from '@testing-library/react-native'
import CreateModel from "../components/CreateModel";
import firebaseConfig from "../firebaseConfig";
import renderer from "react-test-renderer";
import HomeScreen from "../screens/HomeScreen";
import SearchScreen from "../screens/SearchScreen";
import {toMatchImageSnapshot} from "jest-image-snapshot";

const puppeteer = require("puppeteer")

const mockedNavigation = jest.fn();
jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: mockedNavigation,
            dispatch: jest.fn(),
        }),
    };
});

expect.extend({toMatchImageSnapshot});

describe("CreateModal Components", () => {
    let browser;
    jest.setTimeout(35000);

    beforeAll(async () => {
        browser = await puppeteer.launch({args: ['--no-sandbox', '--disable-setuid-sandbox']});
    })

    it('renders correctly', async () => {
        const tree = renderer.create(<CreateModel/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("it makes a screenshot", async () => {
        const page = await browser.newPage();
        await page.goto('http://localhost:19006');

        const openModalSelector = "#root > div > div > div > div > div > div.css-view-1dbjc4n.r-flex-13awgt0 > div > div.css-view-1dbjc4n.r-bottom-1p0dtai.r-display-6koalj.r-left-1d2f490.r-pointerEvents-12vffkv.r-position-u8s1d.r-right-zchlnj.r-top-ipm5af > div.css-view-1dbjc4n.r-flex-13awgt0.r-pointerEvents-12vffkv > div > div > div > div.css-view-1dbjc4n.r-flex-13awgt0 > div > div.css-view-1dbjc4n.r-bottom-1p0dtai.r-left-1d2f490.r-pointerEvents-105ug2t.r-right-zchlnj > div.css-view-1dbjc4n.r-flex-13awgt0.r-flexDirection-18u37iz > a:nth-child(3)";
        await page.waitForSelector(openModalSelector);
        const openModal = await page.$(openModalSelector);
        openModal.click();

        await new Promise((r) => setTimeout(r, 5000));

        const image1 = await page.screenshot();

        const setCategorySelector = "body > div:nth-child(5) > div > div.css-view-1dbjc4n > div > div > div > div.css-view-1dbjc4n.r-flex-kgf08f.r-marginHorizontal-1n9sb9w.r-marginTop-19tq15n > div:nth-child(3) > div.css-view-1dbjc4n.r-alignItems-1awozwy.r-borderBottomWidth-wgabs5.r-borderColor-1lnm5ts.r-cursor-1loqt21.r-flexDirection-18u37iz.r-fontSize-ubezar.r-height-uvuy5l.r-justifyContent-1wtj0ep.r-padding-edyy15.r-touchAction-1otgn73.r-transitionProperty-1i6wzkk.r-userSelect-lrvibr";
        await page.waitForSelector(setCategorySelector);
        const setCategory = await page.$(setCategorySelector);
        setCategory.click();

        await new Promise((r) => setTimeout(r, 5000));

        const image2 = await page.screenshot();

        expect(image1).toMatchImageSnapshot();
        expect(image2).toMatchImageSnapshot();
    })

    it('Title Placeholder is correct', () => {
        const {getByTestId} = render(<CreateModel navigation={{mockedNavigation}}/>);
        const title = getByTestId("CreateModalTextInputTITLE");
        expect(title.props.placeholder).toMatch("e.g. Linear Algebra");
    });

    it('Description Placeholder is correct', () => {
        const {getByTestId} = render(<CreateModel navigation={{mockedNavigation}}/>);
        const title = getByTestId("CreateModalTextInputDESCRIPTION");
        expect(title.props.placeholder).toMatch("Example description");
    });

    it('SelectCategory Modal opens correctly', () => {
        const {getByTestId} = render(<CreateModel navigation={{mockedNavigation}}/>);
        const btn = getByTestId("CreateModalSelectCategoryBTN");
        const modal = getByTestId("SelectCategoryModal");
        expect(modal.props.visible).toEqual(false);
        fireEvent.press(btn);
        expect(modal.props.visible).toEqual(true);
    })

    it('CreateCategory Modal opens correctly', () => {
        const {getByTestId} = render(<CreateModel navigation={{mockedNavigation}}/>);
        const btn = getByTestId("CreateModalCreateCategoryBTN");
        const modal = getByTestId("CreateCategoryModal");
        expect(modal.props.visible).toEqual(false);
        fireEvent.press(btn);
        expect(modal.props.visible).toEqual(true);
    })

    it('Set Border by picking Color correctly', () => {
        const {getByTestId} = render(<CreateModel navigation={{mockedNavigation}}/>);
        const btn = getByTestId("CreateModalPickLEFTColorBTN");
        expect(btn.props.style.borderWidth).toEqual(0);
        fireEvent.press(btn);
        expect(btn.props.style.borderWidth).toEqual(2);
    })
});
