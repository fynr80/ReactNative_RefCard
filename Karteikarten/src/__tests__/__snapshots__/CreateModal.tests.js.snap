// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateModal Components renders correctly 1`] = `
Array [
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    visible={true}
  >
    <RNCSafeAreaView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#add8e6",
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": "2%",
              },
              Object {
                "alignItems": "flex-start",
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 24,
                  "fontWeight": "bold",
                },
                Object {
                  "color": "white",
                  "flex": 4,
                  "height": "100%",
                  "textAlign": "center",
                  "top": "2%",
                },
              ]
            }
          >
            Create a new Set
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "bottom": "-70%",
              "flex": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "white",
                "borderRadius": 150,
                "borderWidth": 10,
                "padding": 10,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 80,
                  "width": 120,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 5,
              "marginHorizontal": 20,
              "marginTop": 80,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "marginVertical": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            Title
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="e.g. Linear Algebra"
            rejectResponderTermination={true}
            style={
              Object {
                "borderBottomWidth": 2,
                "borderColor": "#777",
                "fontSize": 16,
                "height": 60,
                "justifyContent": "space-between",
                "padding": 8,
              }
            }
            testID="CreateModalTextInputTITLE"
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "marginVertical": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            Description
          </Text>
          <TextInput
            allowFontScaling={true}
            multiline={true}
            onChangeText={[Function]}
            placeholder="Example description"
            rejectResponderTermination={true}
            style={
              Object {
                "borderBottomWidth": 2,
                "borderColor": "#777",
                "fontSize": 16,
                "height": 60,
                "justifyContent": "space-between",
                "padding": 8,
              }
            }
            testID="CreateModalTextInputDESCRIPTION"
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "marginVertical": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            Category
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#777",
                "borderRadius": 15,
                "borderWidth": 2,
                "flexDirection": "row",
                "fontSize": 16,
                "height": 60,
                "justifyContent": "space-between",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="CreateModalSelectCategoryBTN"
          >
            <Text>
              select category
            </Text>
            <Text />
          </View>
        </View>
        <View
          style={
            Object {
              "marginVertical": 20,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#FD4365",
                "borderColor": "transparent",
                "borderRadius": 15,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 0,
                "justifyContent": "center",
                "minWidth": 64,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 15,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        Object {
                          "textTransform": "uppercase",
                        },
                        Object {
                          "color": "#6200ee",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 24,
                          "fontWeight": "bold",
                        },
                        Object {
                          "color": "white",
                        },
                      ]
                    }
                  >
                    Create
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </Modal>,
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    testID="SelectCategoryModal"
    visible={false}
  >
    <RNCSafeAreaView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#add8e6",
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": "2%",
              },
              Object {
                "alignItems": "flex-start",
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 24,
                  "fontWeight": "bold",
                },
                Object {
                  "color": "white",
                  "flex": 4,
                  "height": "100%",
                  "textAlign": "center",
                  "top": "2%",
                },
              ]
            }
          >
            Select a Category
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "bottom": "-70%",
              "flex": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "white",
                "borderRadius": 150,
                "borderWidth": 10,
                "padding": 10,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 80,
                  "width": 120,
                }
              }
            />
          </View>
        </View>
      </View>
      <RNCSafeAreaView
        style={
          Object {
            "flex": 5,
            "marginHorizontal": 20,
            "marginTop": 80,
          }
        }
      >
        <RCTScrollView
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </RNCSafeAreaView>
      <View
        style={
          Object {
            "bottom": 20,
            "position": "absolute",
            "right": 20,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 28.5,
                  "height": 57,
                  "width": 57,
                },
                undefined,
                Object {
                  "backgroundColor": "#FD4365",
                },
              ],
            ]
          }
          testID="CreateModalCreateCategoryBTN"
        >
          <View>
            <Text />
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </Modal>,
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    testID="CreateCategoryModal"
    visible={false}
  >
    <RNCSafeAreaView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#add8e6",
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": "2%",
              },
              Object {
                "alignItems": "flex-start",
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 24,
                  "fontWeight": "bold",
                },
                Object {
                  "color": "white",
                  "flex": 4,
                  "height": "100%",
                  "textAlign": "center",
                  "top": "2%",
                },
              ]
            }
          >
            Create a new Category
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "bottom": "-70%",
              "flex": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "white",
                "borderRadius": 150,
                "borderWidth": 10,
                "padding": 10,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 80,
                  "width": 120,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 5,
              "marginHorizontal": 20,
              "marginTop": 80,
            },
          ]
        }
      >
        <View
          style={Object {}}
        >
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "#f6f6f6",
                  "borderColor": "rgba(0, 0, 0, 0.54)",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                Array [
                  Object {
                    "paddingBottom": 0,
                  },
                  Object {
                    "minHeight": 64,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "zIndex": 3,
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#f6f6f6",
                      "bottom": 4,
                      "left": 10,
                      "maxHeight": 2,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                      "width": 12,
                    }
                  }
                />
                <Text
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  style={
                    Object {
                      "backgroundColor": "#f6f6f6",
                      "color": "transparent",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 18,
                      "maxWidth": -28,
                      "opacity": 0,
                      "paddingHorizontal": 0,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 37,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                        Object {
                          "scaleY": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Name
                </Text>
                <Text
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 36,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Name
                </Text>
                <Text
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 36,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Name
                </Text>
              </View>
              <TextInput
                allowFontScaling={true}
                editable={true}
                maxFontSizeMultiplier={1.5}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder=""
                placeholderTextColor="rgba(0, 0, 0, 0.54)"
                rejectResponderTermination={true}
                selectionColor="#6200ee"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 56,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    Array [
                      Object {},
                    ],
                  ]
                }
                testID="text-input-outlined"
                underlineColorAndroid="transparent"
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 20,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FF9C0D",
                "borderRadius": 10,
                "borderWidth": 0,
                "height": "50%",
                "margin": 4,
                "opacity": 1,
                "width": "30%",
              }
            }
            testID="CreateModalPickLEFTColorBTN"
          />
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fe357d",
                "borderRadius": 10,
                "borderWidth": 0,
                "height": "50%",
                "margin": 4,
                "opacity": 1,
                "width": "30%",
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#2B32B2",
                "borderRadius": 10,
                "borderWidth": 0,
                "height": "50%",
                "margin": 4,
                "opacity": 1,
                "width": "30%",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 3,
              "marginTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#FD4365",
                "borderColor": "transparent",
                "borderRadius": 15,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 0,
                "justifyContent": "center",
                "minWidth": 64,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 15,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        Object {
                          "textTransform": "uppercase",
                        },
                        Object {
                          "color": "#6200ee",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 24,
                          "fontWeight": "bold",
                        },
                        Object {
                          "color": "white",
                        },
                      ]
                    }
                  >
                    Create
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </Modal>,
]
`;
